src/main/java/htsjdk/samtools/cram/CRAIEntry.java

```diff
@@ -2,32 +2,41 @@
 
 import htsjdk.samtools.SAMRecord;
 import htsjdk.samtools.cram.structure.Container;
-import htsjdk.samtools.cram.structure.Slice;
+import htsjdk.samtools.cram.structure.slice.SliceBAIMetadata;
 import htsjdk.samtools.util.RuntimeIOException;
 
 import java.io.IOException;
 import java.io.OutputStream;
-import java.util.ArrayList;
-import java.util.Comparator;
-import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * A class representing CRAI index entry: file and alignment offsets for each slice.
  * Created by vadim on 10/08/2015.
  */
-public class CRAIEntry implements Comparable<CRAIEntry>, Cloneable {
```

Comment:
This class is now immutable.

It also does not implement `Cloneable` any more because it no longer has a `clone()` - that was only used by tests.